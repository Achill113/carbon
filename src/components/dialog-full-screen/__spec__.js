import React from 'react';
import { shallow, mount, ReactWrapper } from 'enzyme';
import DialogFullScreen from './dialog-full-screen';
import FullScreenHeading from './full-screen-heading';
import Button from './../button';
import Portal from './../portal';
import { elementsTagTest, rootTagTest } from '../../utils/helpers/tags/tags-specs';
import Icon from './../icon';
import Heading from './../heading';

describe('DialogFullScreen', () => {
  let instance,
      wrapper,
      portal,
      portalContent;
  const onCancel = jasmine.createSpy('cancel');

  beforeEach(() => {
    wrapper = shallow(
      <DialogFullScreen
        onCancel={ onCancel }
        className='foo'
        open
        title='my title'
        subtitle='my subtitle'
      >
        <Button>Button</Button>
        <Button>Button</Button>
      </DialogFullScreen>
    );
    instance = wrapper.instance();
  });

  describe('default props', () => {
    it('sets enableBackgroundUI to true', () => {
      expect(instance.props.enableBackgroundUI).toBeTruthy();
    });
  });

  describe('dialogClasses', () => {
    it('returns the full screen dialog class', () => {
      expect(instance.dialogClasses).toEqual('carbon-dialog-full-screen__dialog');
    });
  });

  describe('mainClasses', () => {
    it('returns the full screen dialog class and custom class', () => {
      expect(instance.mainClasses).toEqual('foo carbon-dialog-full-screen');
    });
  });

  describe('modalHTML', () => {
    beforeEach(() => {
      wrapper = mount(
        <DialogFullScreen
          onCancel={ onCancel }
          className='foo'
          open
          title='my title'
        >
          <Button>Button</Button>
          <Button>Button</Button>
        </DialogFullScreen>
      );
      instance = wrapper.instance();
      portal = wrapper.find(Portal);
      portalContent = new ReactWrapper(portal.node._portal, wrapper);
    });

    it('renders a parent div with mainClasses attached', () => {
      expect(portalContent.find('div.foo.carbon-dialog-full-screen').length).toEqual(1);
    });

    it('renders the dialog', () => {
      expect(instance._dialog).toBeTruthy();
      expect(instance._dialog.className).toEqual('carbon-dialog-full-screen__dialog');
    });

    it('closes when the exit icon is click', () => {
      const closeIcon = portalContent.find(Icon);
      closeIcon.simulate('click');
      expect(onCancel).toHaveBeenCalled();
    });

    it('renders the children passed to it', () => {
      expect(portalContent.find(Button).length).toEqual(2);
    });
  });

  describe('onOpening', () => {
    beforeEach(() => {
      wrapper = mount(<DialogFullScreen open={ false } />);
      wrapper.setProps({ open: true });
    });

    it('adds a carbon-dialog-full-screen--open class to the body', () => {
      const html = wrapper.instance().document.documentElement;
      expect(html.className).toMatch('carbon-dialog-full-screen--open');
    });
  });

  describe('onClosing', () => {
    beforeEach(() => {
      wrapper = mount(<DialogFullScreen open={ false } />);
      wrapper.setProps({ open: true });
    });

    it('removes a carbon-dialog-full-screen--open class to the body', () => {
      const html = wrapper.instance().document.documentElement;
      expect(html.className).toMatch('carbon-dialog-full-screen--open');
      wrapper.setProps({ open: false });
      expect(html.className).not.toMatch('carbon-dialog-full-screen--open');
    });
  });

  describe('dialogTitle', () => {
    describe('is a string', () => {
      it('renders the title within a heading', () => {
        const heading = wrapper.find(Heading);
        expect(heading.props().title).toEqual('my title');
        expect(heading.props().subheader).toEqual('my subtitle');
        expect(heading.props().titleId).toEqual('carbon-dialog-title');
        expect(heading.props().subtitleId).toEqual('carbon-dialog-subtitle');
      });
    });

    describe('is an object', () => {
      beforeEach(() => {
        const titleHeading = <Heading title='my custom heading' />;
        wrapper = shallow(
          <DialogFullScreen
            onCancel={ onCancel }
            className='foo'
            open
            title={ titleHeading }
          >
            <Button>Button</Button>
            <Button>Button</Button>
          </DialogFullScreen>
        );
      });

      it('renders the component in a full screen heading', () => {
        const fullScrenHeading = wrapper.find(FullScreenHeading),
            heading = fullScrenHeading.children().first();

        expect(heading.props().title).toEqual('my custom heading');
      });
    });
  });

  describe('tags', () => {
    describe('on component', () => {
      it('include correct component, elements and role data tags', () => {
        wrapper = shallow(
          <DialogFullScreen
            data-element='bar'
            onCancel={ () => {} }
            onConfirm={ () => {} }
            open
            title='Test'
            data-role='baz'
          />
        );
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
