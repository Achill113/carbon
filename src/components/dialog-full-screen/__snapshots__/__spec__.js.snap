// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogFullScreen tags on component include correct component, elements and role data tags 1`] = `
<Portal
  open={true}
>
  <div
    className="carbon-dialog-full-screen"
    data-component="dialog-full-screen"
    data-element="bar"
    data-role="baz"
  >
    <CSSTransitionGroup
      component="div"
      transitionAppear={false}
      transitionEnter={true}
      transitionEnterTimeout={500}
      transitionLeave={true}
      transitionLeaveTimeout={500}
      transitionName="modal"
    >
      <div
        className="carbon-dialog-full-screen__dialog"
        data-component="dialog-full-screen"
        data-element="bar"
        data-role="baz"
      >
        <Icon
          bgSize="small"
          className="carbon-dialog-full-screen__close"
          data-element="close"
          onClick={[Function]}
          type="close"
        />
        <FullScrenHeading>
          <Heading
            divider={true}
            separator={false}
            subtitleId="carbon-dialog-subtitle"
            title="Test"
            titleId="carbon-dialog-title"
          />
        </FullScrenHeading>
        <div
          className="carbon-dialog-full-screen__content"
          data-element="content"
        />
      </div>
    </CSSTransitionGroup>
    <CSSTransitionGroup
      component="div"
      transitionAppear={false}
      transitionEnter={true}
      transitionEnterTimeout={500}
      transitionLeave={true}
      transitionLeaveTimeout={500}
      transitionName="modal-background"
    />
  </div>
</Portal>
`;
